include/dsn/tool-api/rpc_message.h

```diff
@@ -148,6 +153,41 @@ class message_ex : public ref_counter,
     // by message queuing
     dlink dl;
 
+    // lantency tracer
+    std::unique_ptr<dsn::tool::latency_tracer> tracer;
+
+    void report_tracer(uint64_t time_threshold)
```

Comment:
```suggestion
    std::string report_if_exceed_threshold(uint64_t time_threshold)
```

Formatting the tracing result into a string should be a public function in latency_tracer.
You repeat the work in message and mutation.

Reply:
message.h and mutation.h format result is different, which can't use same formt rule, I  have refactored it